### DISCUSSION TITLE
Autenticação via Cookie

### DISCUSSION THREAD

**Discussion by laurolapajunior** [balta-io/forum#594] [Tira Dúvidas] [answered]
Uma duvida, basicamente fiz o codigo do curso, esta tudo funcionando, em um projeto que estou fazendo, mas o cookie não morre nunca, só entra na tela de login novamente se eu  dar logout, caso contrário sempre entra direto no sistema. eu não cheguei a testar o projeto do curso fullstack se esta assim também. tem algum forma tipo sempre que sair do sistema entrar novamente, tem que  logar.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @laurolapajunior, como vai?

Você pode setar o tempo de expiração do Cookie:
```csharp
builder.Services
            .AddAuthentication(IdentityConstants.ApplicationScheme)
            .AddIdentityCookies(x =>
            {
                x.ApplicationCookie?.Configure(o =>
                {
                    o.Cookie.MaxAge = TimeSpan.FromHours(4);
                    o.ExpireTimeSpan = TimeSpan.FromHours(4);
                });
            });
```

**Reply by maojsm** [reply]
Eu percebi esse problema e realmente não consegui deslogar, resolvi usando as dicas desse link que basicamente remetam ao codio abaixo no BuilderExtensions da API:

[https://andrewlock.net/making-authenticated-cross-origin-requests-with-aspnetcore-identity/](url)

```
builder.Services.Configure<CookieAuthenticationOptions>(
    IdentityConstants.ApplicationScheme,
    options =>
    {
        options.Cookie.SameSite = SameSiteMode.None;
        options.Cookie.SecurePolicy = CookieSecurePolicy.Always; // Garantir que seja apenas HTTPS
    });
```

**Comment by laurolapajunior** [comment]
Entendi, muito obrigado,  e se o usuário estiver usando o sistema? Vai ter
que se logar novamente? Caso sim tem alguma forma de renovar automático?

Em qui., 20 de jun. de 2024 11:42, Andre Baltieri ***@***.***>
escreveu:

**Reply by andrebaltieri** [reply]
Bom dia, @laurolapajunior, como vai?
Tem sim, pode renovar o token (enquanto válido) utilizando uma técnica chamada Refresh Token!

Se notar, nos endpoints dos Identity que geramos nas aulas tem uma URL para isto!
